src/cio_websocket.c

```diff
@@ -461,15 +470,15 @@ static void handle_ping_frame(struct cio_websocket *ws, uint8_t *data, uint_fast
 	}
 
 	ws->ws_private.write_pong_job.wbh = &ws->ws_private.ping_buffer.wb_head;
-	ws->ws_private.write_pong_job.handler = NULL;
+	ws->ws_private.write_pong_job.handler = pong_frame_written;
 	ws->ws_private.write_pong_job.handler_context = NULL;
 	ws->ws_private.write_pong_job.frame_type = CIO_WEBSOCKET_PONG_FRAME;
 	ws->ws_private.write_pong_job.last_frame = true;
 	ws->ws_private.write_pong_job.stream_handler = message_written;
 
 	enum cio_error err = enqueue_job(ws, &ws->ws_private.write_pong_job);
-	if (cio_likely(err == CIO_SUCCESS)) {
-		ws->ws_private.bs->read_exactly(ws->ws_private.bs, ws->ws_private.rb, 1, get_header, ws);
+	if (cio_unlikely(err != CIO_SUCCESS)) {
+		// TODO
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=c%3AS1135)